### PR TITLE
Kmeans

### PR DESCRIPTION
**Pull request by sagarwaghmare69**
Kmeans with Kmeans++ initialization option. Implemented in spirit with torch NN modules, works with gradient descent.

### CODE REVIEW

**Comment by nicholas-leonard** [comment]
@sagarwaghmare69 Looks good. Can you add some documentation? Maybe an example training script?

**Comment by sagarwaghmare69** [comment]
Cool. Yes, I will add documentation.

**Comment by sagarwaghmare69** [comment]
@nicholas-leonard added documentation along with a sample example.

**Comment by sagarwaghmare69** [comment]
@nicholas-leonard I added sample example in the doc itself. Let me know if you would like independent example script.
